### ISSUE TITLE
Travis.genkeys() error - logged in, then fails to log in

### ISSUE THREAD

**Issue by tlycken** [JuliaDocs/Documenter.jl#257]
I'm consistently seing this error when trying to do `Travis.genkeys()` for a newly created package:

```
Successfully logged in as tlycken!
not logged in, please run travis login --pro
ERROR: failed process: Process(`travis encrypt-file .documenter`, ProcessExited(1)) [1]
 in pipeline_error(::Base.Process) at ./process.jl:616
 in run at ./process.jl:592 [inlined]
 in (::Documenter.Travis.##1#2)() at /home/tlycken/.julia/v0.5/Documenter/src/Documenter.jl:467
 in cd(::Documenter.Travis.##1#2, ::String) at ./file.jl:59
 in genkeys(::String) at /home/tlycken/.julia/v0.5/Documenter/src/Documenter.jl:463
```

I can reproduce the behavior with calls to the `travis` utility (which I had never heard of before using this package):

```
$ travis login --auto --org
Successfully logged in as tlycken!
$ travis encrypt-file .documenter
not logged in, please run travis login --pro
```

so the problem is probably not strictly with Documenter.jl, however I'd gladly take any help in troubleshooting. (And maybe Documenter.jl can intercept this and handle it better?)

**Comment by mortenpi** [comment]
Is the repository private by any chance?

**Comment by tlycken** [comment]
Nope, it's a public repo (https://github.com/tlycken/QuickHull.jl).

**Comment by mortenpi** [comment]
I wasn't really able to reproduce it, although I was able to make `travis` fail in other ways. The issue doesn't seem to be Documenter-specific, but generally being more helpful with `travis` failures would probably be good.

I don't have any good ideas right now, but the following information could maybe be useful:

- What OS are you using?
- What is your `travis --version`?
- Do `travis whoami` and `travis status` work after you've logged in?
- What's the `travis endpoint` for QuickHull?
- Is the `travis` command working for other repositories?
- What happens when you run `travis encrypt-file` with `--debug --explode` options?

**Comment by tlycken** [comment]
Thanks for looking into this!

Answers to your questions:

* I'm running this from Ubuntu 16.04
* `travis --version` reports 1.8.2
* `travis whoami` seems to work every time, and correctly identifies me as tlycken. `travis status` works for e.g. `METADATA.jl`, but not for `QuickHull.jl`. I even ran through the following:

        $ travis login --auto --org && travis status       
        Successfully logged in as tlycken!
        not logged in, please run travis login --pro

* The `travis endpoint` is https://api.travis-ci.org

* Not sure what you mean by "the `travis` command". What should I run to test?

* Final question:

        $ travis encrypt-file --debug --explode .documenter
        ** Loading "/home/tlycken/.travis/config.yml"
        ** Timeout::Error: execution expired
        ** Loading gh
        ** GET "config"
        **   took 0.2 seconds
        ** GitHub API: HEAD /repos/tomas.lycken@gmail.com/QHull.jl
        not logged in, please run travis login --pro

**Comment by tlycken** [comment]
Hm. I notice now when I re-read that log output that the url seems wrong. `/repos/tomas.lycken@gmail.com/QHull.jl` should probably be `/repos/tlycken/QuickHull.jl`.

And yes!

Grepping for that string I found that `travis.slug` was set to `tomas.lycken@gmail.com/QHull.jl` in `.git/config` inside this repo. After correcting that setting, everything seems to work as expected.

I caused this myself, by deciding to rename the package by changing a bunch of references manually rather than re-generate the package directory and move the code - so it's probably a pretty rare edge case. But Documenter.jl could probably check this setting and at least emit a warning if it doesn't match the remote of the repo.

Thanks for your help with troubleshooting!
